src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -47,35 +48,22 @@
 }
 
 
-class DeploymentOption:
-    def __init__(self, name: str, available=False, capacity=0, used=0, hdd_used=0,
-                 ssd_used=0, nvme_used=0):
-        self.name = name
-        self.available = available
-        self.capacity = capacity
-        self.used = used
-        self.hdd_used = hdd_used
-        self.ssd_used = ssd_used
-        self.nvme_used = nvme_used
-
-    def as_dict(self):
-        return {
-            'name': self.name,
-            'available': self.available,
-            'capacity': self.capacity,
-            'used': self.used,
-            'hdd_used': self.hdd_used,
-            'ssd_used': self.ssd_used,
-            'nvme_used': self.nvme_used
-        }
-
-
 class DeploymentOptions:
     def __init__(self):
         self.options = {
-            'cost-capacity': DeploymentOption('cost-capacity'),
-            'throughput': DeploymentOption('throughput-optimized'),
-            'iops': DeploymentOption('iops-optimized'),
+            OsdDeploymentOptions.COST_CAPACITY:
+                HostStorageSummary(OsdDeploymentOptions.COST_CAPACITY,
```

Comment:
@nizamial09 Yes I tested this **Cost/Capacity-optimized** option, In this option, we should ask users for **osd encryption** option also currently it's not showing any preview or pop message before deploying osd
So it's good to show this pop-up or preview and confirmation check box from users before creating osd's?

Reply:
Thank you @sunilangadi2 for testing it out!


> we should ask users for **osd encryption** option

You are right. I completely missed that part while implementing it. I'll update the PR shortly



> this pop-up or preview and confirmation check box

I can show the preview but we will be taking it up on the next iteration (separate PR). And if we are showing the preview accurately then I don't think we'll need a confirmation box right?  User saw the preview and verified it before hitting the Confirm. Do we need confirmation box after that?

Reply:
Done!

Reply:
Thanks @nizamial09, But Features under **Encryption option**  is shown but what if user forgot to open it, So I feel it's good to show this **Features** inside each OSD deployment option so that It also confirms that the user-selected required features Before deploying OSD's Please let me know your thoughts? also Preview or confirmation box anyone is fine for me.

Reply:
@sunilangadi2 The `Features` section will be always shown and user cannot hide it. Its an always open accordion.